app/crds.py

```diff
@@ -339,3 +341,62 @@ class TwingateConnectorCRD(BaseK8sModel):
 
 
 # endregion
+
+# region TwingateGroup
+
+
+class GroupSpec(BaseModel):
+    model_config = ConfigDict(
+        frozen=True, populate_by_name=True, alias_generator=to_camel, extra="allow"
+    )
+
+    id: str | None = None
+    name: str | None = None
+    security_policy_id: str | None = None
+    members: set[str] = set()
+
+    @cached_property
+    def members_hash(self) -> str:
+        members_str = ",".join(sorted(self.members))
+        return hashlib.md5(members_str.encode()).hexdigest()  # noqa: S324 # nosec
+
+    @cached_property
+    def members_email(self) -> set[str]:
+        return {v for v in self.members if "@" in v}
+
+    @cached_property
+    def members_ids(self) -> set[str]:
+        return {v for v in self.members if "@" not in v}
+
+
+class GroupStatusUserID(BaseModel):
+    model_config = ConfigDict(
+        frozen=True, populate_by_name=True, alias_generator=to_camel, extra="allow"
+    )
+
+    id: str
+    email: str | None = None
+
+
+class GroupStatus(BaseModel):
+    model_config = ConfigDict(
+        frozen=True, populate_by_name=True, alias_generator=to_camel, extra="allow"
+    )
+
+    user_ids: list[GroupStatusUserID] = []
+    user_ids_hash: str | None = None
```

Comment:
I would rename this to `members_hash` because with this name, I would expect this value to be the has of the `user_ids` list above